### PR TITLE
Fixes incorrect behavior of `yii\validation\NumberValidator` when used with locales where decimal separator is comma

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #10488

### CODE REVIEW

**Comment by rob006** [comment]
Shouldn't this normalize also thousands separator?

**Comment by cebe** [comment]
@rob006 thousand separator is not applied on cast to string afaik.

**Comment by cebe** [comment]
related to #9515

File: framework/validators/NumberValidator.php [hunk 1]
```diff
@@ -117,6 +118,26 @@ protected function validateValue($value)
...
+        $value = "$value";
+
+        $localeInfo = localeconv();
+        $decimalSeparator = isset($localeInfo['decimal_point']) ? $localeInfo['decimal_point'] : null;
+
+        if ($decimalSeparator !== null && $decimalSeparator !== '.') {
+            $value = str_replace($decimalSeparator, '.', $value);
+        }
+
+        return $value;
```
Comment (samdark) [inline framework/validators/NumberValidator.php]: Probably.

**Comment by samdark** [comment]
I have no idea why it fails on travis. Locally it runs just fine :(

**Comment by rob006** [comment]
PHP `setlocale()` depends on locales installed on system. Travis probably does not have `da_DK.UTF-8` locale installed by default.

**Comment by samdark** [comment]
@rob006 yes, that could be it. Any idea of travis locales available?

**Comment by rob006** [comment]
```
$ locale -a
ar_AE.utf8
ar_EG.utf8
C
C.UTF-8
de_AT.utf8
de_BE.utf8
de_CH.utf8
de_DE.utf8
en_AG
en_AG.utf8
en_AU.utf8
en_BW.utf8
en_CA.utf8
en_DK.utf8
en_GB.utf8
en_HK.utf8
en_IE.utf8
en_IN
en_IN.utf8
en_NG
en_NG.utf8
en_NZ.utf8
en_PH.utf8
en_SG.utf8
en_US.utf8
en_ZA.utf8
en_ZM
en_ZM.utf8
en_ZW.utf8
es_ES.utf8
es_MX.utf8
fr_BE.utf8
fr_CH.utf8
fr_FR.utf8
he_IL.utf8
hi_IN
hi_IN.utf8
ja_JP.utf8
ko_KR.utf8
ms_MY.utf8
POSIX
pt_BR.utf8
pt_PT.utf8
ru_RU.utf8
ru_UA.utf8
uk_UA.utf8
zh_CN.utf8
zh_TW.utf8
```

Similar tests in Symfony: https://github.com/symfony/yaml/blob/master/Tests/InlineTest.php#L96

**Comment by samdark** [comment]
@rob006 failed :(

**Comment by rob006** [comment]
@samdark #13227

File: tests/framework/validators/NumberValidatorTest.php [hunk 2]
```diff
@@ -37,7 +69,13 @@ public function testValidateValueSimple()
         $this->assertTrue($val->validate(-20));
         $this->assertTrue($val->validate('20'));
         $this->assertTrue($val->validate(25.45));
+
+        $this->setPointDecimalLocale();
         $this->assertFalse($val->validate('25,45'));
+        $this->setCommaDecimalLocale();
+        $this->assertTrue($val->validate('25,45'));
```
Comment (cebe) [inline tests/framework/validators/NumberValidatorTest.php]: did this pass before? what if the value if passed to a database which does not understand `,` as decimal separator?

Comment (samdark) [inline tests/framework/validators/NumberValidatorTest.php]: Values conversion is a separate issue. Here we're talking about user input only which is always done according to country tradition and is always considered valid for that particular country.

**Pull request merged**